src/sandstorm/run-bundle.c++

```diff
@@ -2282,29 +2282,27 @@ private:
   }
 
   void mongoCommand(const Config& config, kj::StringPtr command, kj::StringPtr db = "meteor") {
-    pid_t pid = fork();
-    if (pid == 0) {
-      // We don't want to unwind the stack in this subprocess.
-      KJ_IF_MAYBE(exception, kj::runCatchingExceptions([&]() {
-        // Don't run as root.
-        dropPrivs(config.uids);
-
-        execMongoClient(config, {"--quiet", "--eval", command }, db);
-      })) {
-        context.exitError(kj::str(*exception));
-      }
+    char commandFile[] = "/tmp/mongo-command.XXXXXX";
+    int commandFd;
+    KJ_SYSCALL(commandFd = mkstemp(commandFile));
+    KJ_DEFER(unlink(commandFile));
+    KJ_SYSCALL(fchown(commandFd, -1, config.uids.gid));
+    KJ_SYSCALL(fchmod(commandFd, 0660));
+    kj::FdOutputStream(kj::AutoCloseFd(commandFd)).write(command.begin(), command.size());
```

Comment:
Exception safety: It's important that `commandFd` be claimed by an `AutoCloseFd` immediately after the fd is created, before anything else that could possibly throw is executed. Otherwise the FD will leak on exceptions.